### PR TITLE
Added a Research Objectives Screen

### PR DESCRIPTION
**Pull request by ThePiachu**
## About The Pull Request

I've played a bit of Research and noticed how disorganised it feels at times. There really isn't an easy way of tracking which chems were done and which you still need to synthesize / analyze. So I figured I'd give Research a similar objectives screen to IOs:

image

At the moment it's a bit sparse, but ideally we would have more objectives for Research to do in the future so we could expand the screen as needed.

The screen doesn't really reveal any information Research wouldn't know, asides from the names of unidentified chems before they are located which isn't really all useful by itself.

Also cleaned the code around chems. Moved two global variables into the Chemical Data struct, added one around tracking unfinished chems, and fixed a piece of code that didn't properly check for chem flags (it had `&&` instead of `&` and would reject all generated chems).

## Why It's Good For The Game

It helps Research stay organised and know which chems they still need to work on.

## Changelog

:cl:
add: Added a View Research Objective verb. This will let Researchers be able to view which chems they still need to synthesize and analyze to keep more organized.
fix: Fixed the chem code that saves researchable chems. It used not to save custom chems, now it does.
code: Cleaned up the code around tracking chems and chem objectives a little. It now actually tracks which chems haven't been researched.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/ResearchMemories.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { useBackend, useLocalState } from '../backend';
+import {
+  Button,
+  Flex,
+  Section,
+  Box,
+  Tabs,
+  Fragment,
```
Comment (harryob) [inline tgui/packages/tgui/interfaces/ResearchMemories.js:8]: don't need to import button and fragment if you're not using them

File: tgui/packages/tgui/interfaces/ResearchMemories.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  const { clearance, research_credits, theme, clue_categories } = data;
+
+  return (
+    <Window width={650} height={700} theme={theme}>
+      <Window.Content scrollable>
+        <Section title={'Clearance: ' + clearance}>
+        </Section>
+        <Section title={'Research Credits: ' + research_credits}>
+        </Section>
```
Comment (harryob) [inline tgui/packages/tgui/interfaces/ResearchMemories.js:29]: empty fragments autoclose, linter fail. you might wanna use something other than sections for this though

**Review by harryob — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/ResearchMemories.js (hunk 2 above)
Comment (ThePiachu) [reply tgui/packages/tgui/interfaces/ResearchMemories.js:29]: What would you suggest? I'm really bad at JS...

Comment (ThePiachu) [reply tgui/packages/tgui/interfaces/ResearchMemories.js]: Changed it to Flex.Item

**Comment by ThePiachu** [comment]
Okay, tweaked the screen a touch based on the feedback:
image
(moved the Research Credits into a smaller sub-section)

**Review by fira — APPROVED** [review]

**Pull request merged**